### DISCUSSION TITLE
Deta Dirve CSV file into dataframe

### DISCUSSION THREAD

**Discussion by dlink1999** [deta/cloud-docs#234] [Help]
Dear Sir: Sorry I am quite new in fastapi and deta.
May I ask a very simple and naive question?
After get a csv file (presumly as txt file?) from Drive, how can I convert it into dataframe? Is there any reference code? Thanks. 

@app.get("/download/{name}")
def read_file(name: str):
    file_txt = drive.get(name)

**Comment by jnsougata** [comment]
Ig you are using `pandas` ... `pd.read_csv(filepath_or_buffer)` .... so you can do like 
```python
@app.get("/download/{name}")
def read_file(name: str):
    file_txt = drive.get(name)
    buff = io.BytesIO(file_txt)
    df = pd.read_csv(buff)
``` 
assuming `file_txt` is of type `bytes`
reference: https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html#pandas-read-csv

**Reply by dlink1999** [reply]
Many thanks. I'll give it a try.
